examples/load_balancer/load_balancer.c

```diff
@@ -143,19 +141,43 @@ usage(const char *progname) {
  */
 static int
 parse_app_args(int argc, char *argv[], const char *progname) {
-        int c;
+        int c, ret;
+        char *client_args, *server_args;
+        char client_ip[32], server_ip[32];
 
         lb->cfg_filename = NULL;
-        lb->client_iface_name = NULL;
-        lb->server_iface_name = NULL;
+        lb->client_port = RTE_MAX_ETHPORTS+1;
+        lb->server_port = RTE_MAX_ETHPORTS+1;
 
         while ((c = getopt(argc, argv, "c:s:f:p:")) != -1) {
                 switch (c) {
                         case 'c':
-                                lb->client_iface_name = strdup(optarg);
+                                client_args = strdup(optarg);
+                                ret = sscanf(client_args, "%[^/]/%hhd", client_ip, &lb->client_port);
+                                if (ret != 2) {
+                                        RTE_LOG(INFO, APP, "Load balancer NF requires a client interface IP and port ID, separated by a /.\n");
+                                        return -1;
+                                }
+                                ret = onvm_pkt_parse_ip(client_ip, &lb->ip_lb_client);
+                                if (ret < 0) {
+                                        RTE_LOG(INFO, APP, "Error parsing client IP address");
+                                        return -1;
+                                }
+                                lb->ip_lb_client = rte_cpu_to_be_32(lb->ip_lb_client);
                                 break;
                         case 's':
-                                lb->server_iface_name = strdup(optarg);
+                                server_args = strdup(optarg);
```

Comment:
This duplicates the client arg parsing, I'd consider making this logic into its own function.

Reply:
Adjusted